File: go/exercise34/employee_list_removal.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"slices"
)

var employees = []string{
	"John Smith",
	"Jackie Jackson",
	"Chris Jones",
	"Amanda Cullen",
	"Jeremy Goodwin",
}

func main() {
	scanner := bufio.NewScanner(os.Stdin)

	displayEmployees(employees)

	fmt.Print("\nEnter an employee name to remove: ")
	scanner.Scan()
	name := scanner.Text()

	if !slices.Contains(employees, name) {
		fmt.Printf("the name %q does not exists\n", name)
		os.Exit(0)
	}

	employees = slices.DeleteFunc(employees, func(emp string) bool {
		return emp == name
	})

	displayEmployees(employees)
}

func displayEmployees(names []string) {
	fmt.Printf("There are %d employees:\n", len(names))

	for _, name := range names {
		fmt.Printf("  - %s\n", name)
	}
}

File: c/exercise34/employee_list_removal.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

void remove_employee(char *, char *);

char *names[] = {"John Smith", "Jackie Jackson", "Chris Jones", "Amanda Cullen",
                 "Jeremy Goodwin"};

int main(void) {
  char remove[15];

  int names_length = (int)(sizeof(names) / sizeof(names[0]));

  printf("There are %d employees:\n", names_length);
  for (int i = 0; i < names_length; i++) {
    printf("- %s\n", names[i]);
  }

  printf("Enter an employee name to remove: ");
  scanf("%s", remove);

  return EXIT_SUCCESS;
}

void remove_employee(char *array, char *name) {
  int names_length = (int)(sizeof(names) / sizeof(names[0]));

  for (int i = 0; i < names_length; i++) {
  }
}
